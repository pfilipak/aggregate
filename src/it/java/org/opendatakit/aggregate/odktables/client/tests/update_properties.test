# Set up
createUser user1
createUser user2

createSynchronizedTable user1 people {'key':'0'}
    name STRING false {'key':'0'}
    age INTEGER false {'key':'0'}
    weight DECIMAL false {'key':'0'}

setTablePermissions user1 user2 people true false false

# Test update table properties
updateTableProps user1 people {'key':'1'}
    
printTable user1 people
    
# Test user can't update properties of non-existent table
updateTableProps user1 nosuchtable {'key':'xxxx'}

# Test user without permission can't update properties
cloneSynchronizedTable user2 people

updateTableProps user2 people {'key':'xxxx'}

# Test user with permission can update properties
setTablePermissions user1 user2 people true true false

updateTableProps user2 people {'key':'xxxx'}

printTable user2 people

removeTableSynchronization user2 people

# Test updating column properties
updateColumnProps user1 people
    name {'key':'1'}
    age {'key':'1'}
	
printTable user1 people

# Test can't update column properties without permission
setTablePermissions user1 user2 people true false false

cloneSynchronizedTable user2 people

updateColumnProps user2 people
    name {'key':'1'}
    
# Test can update properties with permission
setTablePermissions user1 user2 people true true false

updateColumnProps user2 people
    name {'key':'1'}
    
printTable user2 people
    
removeTableSynchronization user2 people

# Test can't update column properties of non-existent columns
updateColumnProps user1 people
    nosuchcol {'key':'xxxx'}
    
# Test can't update column properties of non-existent table
updateColumnProps user1 nosuchtable
    name {'key':'xxxx'}

# Clean up
setTablePermissions user1 user2 people false false false
deleteSynchronizedTable user1 people
deleteUser user1
deleteUser user2