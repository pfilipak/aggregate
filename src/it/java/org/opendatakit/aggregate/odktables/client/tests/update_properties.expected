# Set up
createUser user1
createUser user2

createSynchronizedTable user1 people {'key':'0'}

setTablePermissions user1 user2 people true false false

# Test update table properties
updateTableProps user1 people {'key':'1'}
    
printTable user1 people
table properties: {'key':'1'}
column properties:
    age {'key':'0'}
    name {'key':'0'}
    weight {'key':'0'}
modification: 0
    
# Test user can't update properties of non-existent table
updateTableProps user1 nosuchtable {'key':'xxxx'}
Exception: org.opendatakit.aggregate.odktables.client.exception.TableDoesNotExistException: Table with ID nosuchtable does not exist!

# Test user without permission can't update properties
cloneSynchronizedTable user2 people

updateTableProps user2 people {'key':'xxxx'}
Exception: org.opendatakit.aggregate.odktables.client.exception.PermissionDeniedException: Requesting user does not have permission for that operation!

# Test user with permission can update properties
setTablePermissions user1 user2 people true true false

updateTableProps user2 people {'key':'xxxx'}

printTable user2 people
table properties: {'key':'xxxx'}
column properties:
    age {'key':'0'}
    name {'key':'0'}
    weight {'key':'0'}
modification: 0

removeTableSynchronization user2 people

# Test updating column properties
updateColumnProps user1 people
	
printTable user1 people
table properties: {'key':'1'}
column properties:
    age {'key':'1'}
    name {'key':'1'}
    weight {'key':'0'}
modification: 0

# Test can't update column properties without permission
setTablePermissions user1 user2 people true false false

cloneSynchronizedTable user2 people

updateColumnProps user2 people
Exception: org.opendatakit.aggregate.odktables.client.exception.PermissionDeniedException: Requesting user does not have permission for that operation!
    
# Test can update properties with permission
setTablePermissions user1 user2 people true true false

updateColumnProps user2 people
    
printTable user2 people
table properties: {'key':'xxxx'}
column properties:
    age {'key':'1'}
    name {'key':'1'}
    weight {'key':'0'}
modification: 0
    
removeTableSynchronization user2 people

# Test can't update column properties of non-existent columns
updateColumnProps user1 people
Exception: org.opendatakit.aggregate.odktables.client.exception.ColumnDoesNotExistException: Column with name nosuchcol does not exist in table with tableID people
    
# Test can't update column properties of non-existent table
updateColumnProps user1 nosuchtable
Exception: org.opendatakit.aggregate.odktables.client.exception.TableDoesNotExistException: Table with ID nosuchtable does not exist!

# Clean up
setTablePermissions user1 user2 people false false false
deleteSynchronizedTable user1 people
deleteUser user1
deleteUser user2